Sources/Fluent/Database/ThreadConnectionPool.swift

```diff
@@ -2,87 +2,125 @@ import Foundation
 import Core
 
 /**
-    Responsible for maintaing of pool
+    Responsible for maintaing a pool
     of connections, one for each thread.
- 
-    Uses a Driver to make new connections.
 */
 public final class ThreadConnectionPool {
-    var makeConnection: ConnectionFactory
-    var storage: [pthread_t: (conn: Connection, time: Int)]
-    var maxConnections: Int
-    var time: Int
-    var storageLock: Lock
-    
+
+    /**
+        Thread Pool Errors
+    */
+    public enum Error: Swift.Error {
+        /**
+            Something in our internal lock mechanism has unexpectedly failed 
+            ... should never see this except for more widespread system 
+            dispatch errors
+        */
+        case lockFailure
+
+        /**
+            The maximum number of active connections has been reached and the pool
+            is no longer capable of creating new ones.
+        */
+        case maxConnectionsReached(max: Int)
+
+        /**
+            This is here to allow extensibility w/o breaking apis, it is not currently
+            used, but should be accounted for by end user if they are handling the 
+            error
+        */
+        case open(Swift.Error)
```

Comment:
I'm on the fence about the name `.open`. If I were using this framework and I saw that in a switch statement I would think it has something to do with opening connections. `.unknown` is unambiguous.